WordPress/gutenberg
# Home Link: Render Home text if there is no attribute label present

Fixes https://github.com/WordPress/gutenberg/issues/58307

## What?


Renders default label as "Home" rather than using `useEffect` to set the attribute in the Editor client side unless user explicitly sets it on the block themselves and saves it.

## Why?


Using the Block Hooks API it's now possible to insert this block into the Navigation without default attributes since https://github.com/WordPress/gutenberg/pull/57754.

As it stands, without a default `label` attribute this block won't render on the frontend.

## How?


This renders a default label as `Home` unless a user has set this attribute, in which case it will override the default. We are unable to set the default in `block.json` since it won't be translated.

## Testing Instructions

1. Add the below code to the `block.json` file of the Home Link block.
2. Load this in the Editor and check it's auto-inserted and the link renders as "Home".
3. Do the same for the frontend.
4. Change the text in the Editor and save.
5. Check this is reflected in the Editor upon reload, and again on the frontend.


```json
// packages/block-library/src/home-link/block.json

"blockHooks": {
	"core/navigation": "lastChild"
}
```

### Testing Instructions for Keyboard


## Screenshots or screencast
